src/librustc_trait_selection/traits/select/candidate_assembly.rs

```diff
@@ -145,7 +145,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
 
     fn assemble_candidates_from_projected_tys(
         &mut self,
-        obligation: &TraitObligation<'tcx>,
+        obligation: &PolyTraitObligation<'tcx>,
```

Comment:
This is a curious change -- I'm not quite sure what is driving it? As I said earlier, I expect *eventually* that candidate selection will be and large be operating on non-poly things (but maybe not in this PR)

Reply:
I changed the type definition of `TraitObligation` to be without a binder and added `PolyTraitObligation` for cases where a binder is used.

So this is only a rename, not an actual change